sanic/pages/directory_page.py

```diff
@@ -0,0 +1,59 @@
+from typing import Iterable, TypedDict
+
+from html5tagger import E
+
+from .base import BasePage
+
+
+class FileInfo(TypedDict):
+    icon: str
+    file_name: str
+    file_access: str
+    file_size: str
+
+
+class DirectoryPage(BasePage):  # no cov
+    TITLE = "Directory Viewer"
+
+    def __init__(
+        self, files: Iterable[FileInfo], url: str, debug: bool
+    ) -> None:
+        super().__init__(debug)
+        self.files = files
+        self.url = url
+
+    def _body(self) -> None:
+        with self.doc.main:
+            self._headline()
+            files = list(self.files)
+            if files:
+                self._file_table(files)
+            else:
+                self.doc.p("The folder is empty.")
+
+    def _headline(self):
+        """Implement a heading with the current path, combined with
+        breadcrumb links"""
+        with self.doc.h1(id="breadcrumbs"):
+            p = self.url.split("/")[:-1]
+
+            for i, part in enumerate(p):
+                if i == 0:
+                    part = "🏠"
+                path = "/".join(p[: i + 1]) + "/"
+                self.doc.a(part, href=path)("/")
```

Comment:
```suggestion
                with self.doc.a(href=path):
                    self.doc.span(part, class_="dir").span("/", class_="sep")
```
You should be able to then style it in CSS the way you want it, but slashes are clickable.